x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouteDependencies } from '../../plugin';
+import { createEnterpriseSearchRequestHandler } from '../../lib/enterprise_search_request_handler';
+
+interface ICredentialsResponse {
+  results: object[];
```

Comment:
Should we get more specific with this type def? I'm seeing this example in the tests:

```
              id: 'loco_moco_account_id:5f3575de2b76ff13405f3155|name:asdfasdf',
              key: 'search-fe49u2z8d5gvf9s4ekda2ad4',
              name: 'asdfasdf',
              type: 'search',
              access_all_engines: true, 
```

So we could do something like this:

```suggestion
interface ICredential {
  id: string;
  key: string;
  name: string;
  type: string;
  access_all_engines: boolean;
}
interface ICredentialsResponse { 
  results: ICredential[];
```

Or is that too overkill? No worries if so, I don't feel super strongly about it

Reply:
After having poked at this / tried to use this in other routes, I think I see why it basically only exists for the `hasValidFunc` typing. Feel free to disregard this comment if you'd rather not type out our API responses

Reply:
Let's go with it, we'll end up needed ICredential types elsewhere I think anyway.